private/bufpkg/bufmodule/bufmodule.go

```diff
@@ -421,7 +421,11 @@ func ModuleDigestB3(ctx context.Context, module Module) (string, error) {
 			return "", err
 		}
 	}
-	// TODO: Should we include module.DocumentationPath here?
+	if docPath := module.DocumentationPath(); docPath != "" {
```

Comment:
I do think for compatibility if `buf.md` was the file name, that's the behavior we had before, so it shouldn't update the b3 digest.

I think to address this, we'd change the conditional to be `docPath != "" && dockPath != "buf.md"` (w/ the constant for "buf.md").

Reply:
but then wouldn't it skip updating the digest if the filename was changed from one of the `README`'s to `buf.md`?

Reply:
We never looked at the other filenames for docs before this PR, so that should be safe.

Reply:
I mean going forward, what if a module's filename is changed to `README.md` and then back to `buf.md`? wouldn't `buf.md` be excluded from the digest due to `docPath != "" && docPath != "buf.md"`?

Reply:
Yeah - in that case the documentation path won't be part of the digest, so the digest will still change.

What I was trying to avoid here is having the b3 digest change if the buf.md file was always used.

Reply:
ah you are right, the digest will still change. I guess whether the documentation path is included or not isn't very important in this case as long as the digests are different?

Reply:
@pkwarren is there a reason for why we're avoiding the digest changing? Is it because we'll cache invalidate basically everything, or are checksums no longer going to match (i.e., we can never change the digest function for a module that has already been published)?
To me it seems odd that the docs are _sometimes_ included in the manifest and _sometimes_ are not (based on the filename).

Reply:
We use the b3 digest server side to determine if we should short circuit and prevent an identical push. If we changed the hash here we'd allow a new push for a module that hadn't changed. This might also lead to a ping/pong effect pushing to the same module from two clients with different cli versions.

Reply:
Don't we still have that problem with this PR? If you have a `README.md`, your module's hash is now computed differently once this lands (CLI vNext), and this can result in ping/pong with older CLI versions.

Reply:
Yeah but it is probably more likely someone had a buf.md file before this change since that was the only supported doc path.

Reply:
What about the scenario where someone didn't have a `buf.md`, but now wants to add one? The BSR won't accept the push because the digest didn't change. Granted we're in this problem today, it seems, but it's definitely undesirable behaviour.

I feel like it's reasonable to require users to use a consistent version of the buf CLI at push time, (i.e., in CI/Github Action/GitHub App).

Reply:
The bsr would accept that change because GetDocumentation() would change from empty to the contents of the buf.md file.

Reply:
So with this change, we have the behavior:

* buf push (1.17.0), buf.md docs: no changes
* buf push (1.18.0), buf.md docs: no changes

* buf push (1.17.0), README.* docs: no changes (doesn't support new path)
* buf push (1.18.0), README.* docs: changes (pushes new path and docs)

If we leave this change out, then identical modules (buf.md w/ same contents as before) will result in duplicate pushes to the BSR with no changes between them.

Reply:
Oh I'm totally misreading this, it's the _path_ that we are debating hashing, not the docs themselves. 

Is the issue that we'll have a duplicate push to the BSR with no changes, or ping-pong? Ping-pong I see as less of a concern, as users will get strange behaviour if they are not consistent in the version of the CLI they use (e.g., in this case, docs appearing and disappearing and reappearing).

Reply:
Agree as users will see that with new functionality. I just was trying to prevent pushing identical commits and maybe the likelihood of ping/pong effects would be higher for modules that have a buf.md file today.

Reply:
Thank you both for reasoning through this again 🙏